sycl/test-e2e/Matrix/joint_matrix_opt_kernel_feature_unsupported_hw.cpp

```diff
@@ -1,5 +1,6 @@
 // REQUIRES: gpu-intel-gen12
-// TODO: Remove unsupported after fixing https://github.com/intel/llvm/issues/12683
+// TODO: Remove unsupported after fixing
+// https://github.com/intel/llvm/issues/12683
 // UNSUPPORTED: accelerator
```

Comment:
Does it mean we can remove "// UNSUPPORTED: accelerator" here?

Reply:
IIUC, this test failed on opencl:fpga with the following error. That's why I added "UNSUPPORTED: accelerator". I'm not sure why you are suggesting removing this?

```
2024-02-09T00:53:18.1728039Z env ONEAPI_DEVICE_SELECTOR=opencl:fpga  /__w/llvm/llvm/build-e2e/Matrix/Output/joint_matrix_opt_kernel_feature_unsupported_hw.cpp.tmp.out
2024-02-09T00:53:18.1730184Z # executed command: env ONEAPI_DEVICE_SELECTOR=opencl:fpga /__w/llvm/llvm/build-e2e/Matrix/Output/joint_matrix_opt_kernel_feature_unsupported_hw.cpp.tmp.out
2024-02-09T00:53:18.1731519Z # .---command stderr------------
2024-02-09T00:53:18.1734608Z # | joint_matrix_opt_kernel_feature_unsupported_hw.cpp.tmp.out: /__w/llvm/llvm/llvm/sycl/test-e2e/Matrix/joint_matrix_opt_kernel_feature_unsupported_hw.cpp:27: int main(): Assertion `(e.code() == sycl::errc::kernel_not_supported) && (std::string(e.what()) == std::string("no matrix hardware on the target device, joint_matrix " "is not supported"))' failed.
2024-02-09T00:53:18.1737560Z # `-----------------------------
```

Reply:
I thought your PR fixes https://github.com/intel/llvm/issues/12683
which is why we added unsupported for FPGA.
Adding @dm-vodopyanov since he is the one who added unsupported in the first place for this test.

Reply:
This test is supposed to be run on HW which doesn't support Joint Matrix and verifies that correct exception was generated. So, I would expect that UNSUPPORTED should not be required for accelerator.

Reply:
I see. So, should we add XFAIL for FPGA instead?

Reply:
We need @dm-vodopyanov to look at why it is failing as the author of the feature/test.

Reply:
Hi @uditagarwal97, I don't fully understand, why the test even launched on opencl:fpga if we have `// REQUIRES: gpu-intel-gen12`?
The rationale of this test is to be executed only on Gen12 and to make sure that the expected exception is thrown. No test updates (like the update here) should be made as it is expected that test does not support other devices, especially FPGA which is not a target for the matrices.

@aelovikov-intel having `// REQUIRES: gpu-vendor-abc` in test source means that 
- the test will be executed only if GPU on the testing machine is vendor-abc AND the test will be executed **only on GPU device**, 
- or the test will be executed only if GPU on the testing machine is vendor-abc AND will be executed **on all available devices**?

Reply:
It's passed like `llvm-lit --param gpu-vendor-abc=True` and that sets the feature unconditionally for all the devices (not handled at the `format.py` level). If one wants to limit to `gpu` only, an additional `REQUIRES: gpu` is needed.

Reply:
Thank you @aelovikov-intel.
@uditagarwal97: can you please remove these lines 2-4 added in this patch and modify the line 1 by adding `, gpu` to it?

Reply:
Sure